grpclb/src/main/java/io/grpc/grpclb/GrpclbState.java

```diff
@@ -245,6 +356,7 @@ public void run() {
       // stream.
       loadRecorder = new GrpclbClientLoadRecorder(time);
       lbRequestWriter = stub.withWaitForReady().balanceLoad(this);
```

Comment:
Is this safe to call `balanceLoad(this)` before the constructor is finished? Is it better to do RPC call after `LbStream` is `new()`ed?

Reply:
It's totally safe. Normally `balancerLoad(this)` should not throw, unless there is a bug. If that happens, the exception will be caught by `ChannelExecutor`, and will hopefully be propagated in a more useful way once #3293 is resolved.

Reply:
My point is the RPC may call methods of `this`, while `this` is not fully initialize, for example: `loadReportingTask` is null at this point, so `this.onNext()` may throw NPE if the RPC gets a response before a request being sent. So moving the line `loadReportingTask = new LoadReportingTask(this)` above the RPC can fix it, but this fix is based on the assumption you know how RPC may misbehave.

Reply:
I see. I have moved the lines that reference into `start()`, and create `LoadReportingTask` every time it's scheduled instead of re-using one.